integrations/access/opsgenie/app.go

```diff
@@ -17,15 +17,573 @@ limitations under the License.
 package opsgenie
 
 import (
-	"github.com/gravitational/teleport/integrations/access/common"
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/gravitational/trace"
+	"github.com/jonboulle/clockwork"
+	"google.golang.org/grpc"
+	grpcbackoff "google.golang.org/grpc/backoff"
+
+	tp "github.com/gravitational/teleport"
+	"github.com/gravitational/teleport/api/client"
+	"github.com/gravitational/teleport/api/client/proto"
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/integrations/access/common/teleport"
+	"github.com/gravitational/teleport/integrations/lib"
+	"github.com/gravitational/teleport/integrations/lib/backoff"
+	"github.com/gravitational/teleport/integrations/lib/credentials"
+	"github.com/gravitational/teleport/integrations/lib/logger"
+	"github.com/gravitational/teleport/integrations/lib/watcherjob"
 )
 
 const (
-	// opsgeniePluginName is used to tag Opsgenie GenericPluginData and as a Delegator in Audit log.
-	opsgeniePluginName = "opsgenie"
+	// pluginName is used to tag Opsgenie GenericPluginData and as a Delegator in Audit log.
+	pluginName = "opsgenie"
+	// minServerVersion is the minimal teleport version the plugin supports.
+	minServerVersion = "6.1.0"
+	// grpcBackoffMaxDelay is a maximum time GRPC client waits before reconnection attempt.
+	grpcBackoffMaxDelay = time.Second * 2
+	// initTimeout is used to bound execution time of health check and teleport version check.
+	initTimeout = time.Second * 10
+	// handlerTimeout is used to bound the execution time of watcher event handler.
+	handlerTimeout = time.Second * 5
+	// modifyPluginDataBackoffBase is an initial (minimum) backoff value.
+	modifyPluginDataBackoffBase = time.Millisecond
+	// modifyPluginDataBackoffMax is a backoff threshold
+	modifyPluginDataBackoffMax = time.Second
 )
 
+// errMissingAnnotation is used for cases where request annotations are not set
+var errMissingAnnotation = errors.New("")
```

Comment:
Even if this error is always ignored, we should include a string message in it anyway. A future mistake could lead to this not being ignored and instead being propagated, and an empty string would make this very hard to debug.